apps/aehttp/src/aehttp_dispatch_ext.erl

```diff
@@ -244,15 +244,18 @@ handle_request('GetAccountBalance', Req, _Context) ->
 handle_request('GetCommitmentHash', Req, _Context) ->
     Name         = maps:get('name', Req),
     Salt         = maps:get('salt', Req),
-    CHash        = aens:get_commitment_hash(Name, Salt),
-    EncodedCHash = aec_base58c:encode(commitment, CHash),
-    {200, [], #{commitment => EncodedCHash}};
+    case aens:get_commitment_hash(Name, Salt) of
+        {ok, CHash} ->
+            EncodedCHash = aec_base58c:encode(commitment, CHash),
+            {200, [], #{commitment => EncodedCHash}};
+        {error, Reason} ->
+            ReasonBin = atom_to_binary(Reason, utf8),
```

Comment:
This is relying on undocumented/unspec-ed `aens:get_commitment_hash` return type.

Reply:
Yes, how do we want to document that? Swagger? Erlang spec?

Reply:
My point is that is this code path is exercised then `Reason` may be not an atom and this line will cause client to see 500 rather than 400.

So the point is: `can aens:get_commitment_hash` be spec-ed so that it returns `{error, atom()}`? If not, then this code path shall not use atom_to_binary but rather some pretty-printing.

Reply:
I see, agreed, I'll change that specs to `atom().`